src/dos/dos_locale.cpp

```diff
@@ -2616,34 +2617,23 @@ static void refresh_currency_format(const LocaleInfoEntry &source)
 
 	bool found = false;
 	for (const auto& candidate_utf8 : source.currency_symbols_utf8) {
-		std::string candidate = {};
 
 		// Check if the currency can be converted to current code page
+		const auto candidate = utf8_to_dos(candidate_utf8,
+		                                   DosStringConvertMode::NoSpecialCharacters,
+		                                   UnicodeFallback::Null);
 
-		if (!utf8_to_dos(candidate_utf8,
-		                 candidate,
-		                 DosStringConvertMode::NoSpecialCharacters,
-		                 UnicodeFallback::Null)) {
-			continue;
-		}
-		if (candidate.length() > MaxCurrencySymbolLength) {
+		// TODO[C++23]: Use 'contains()' instead of 'find()'
+		if (candidate.find('\0') != std::string::npos ||
+		    candidate.length() > MaxCurrencySymbolLength) {
 			continue;
```

Comment:
Usage case: discards the string if `Null` fallback fails.  Also looks confusing.  Why are we searching for a null termination character in the middle of a C++  `std::string`?

Reply:
As above, changed the fallback mode.